fix(correlation): add lower bounds for selected events

hazzadous opened the pull request:
Previously we would consider all events for correlation calculation. Now
we use the funnel `date_from` as the lower bounds.

Commit:
fix(correlation): ensure correlation partitioned by team

Previously if distinct_id's were the same between two teams, we'd end up
pulling in the event data between the teams.

Commit:
fix(correlation): add lower bounds for selected events

Previously we would consider all events for correlation calculation. Now
we use the funnel `date_from` as the lower bounds.

Commit:
dev(tests): clear cache between partition calls

Commit:
chore(correlation): exclude funnel steps

Commit:
chore(correlation): make sure cache is cleared before each test

neilkakkar commented on ee/clickhouse/queries/funnels/funnel_correlation.py:
```diff
@@ -151,46 +151,107 @@ def get_partial_event_contingency_tables(self) -> List[EventContingencyTable]:
-                -- funnel (I think)
-                countIf(person.steps = target_step) AS success_count,
+                -- If we have a `person.steps = target_step`, we know the person 
+                -- reached the end of the funnel (I think)
```
```suggestion
                -- reached the end of the funnel
```

hazzadous commented on ee/clickhouse/queries/funnels/funnel_correlation.py:
```diff
@@ -151,46 +151,107 @@ def get_partial_event_contingency_tables(self) -> List[EventContingencyTable]:
+                                person_id,
+                                max(_timestamp) as _timestamp
+                            FROM person_distinct_id
+                            WHERE team_id = {self._team.pk}
```
How might I know that this is available btw?

tbh I think it's more understandable like this rather than relying on the params that are coming from another class. The only think I think we should rely on `funnel_persons_params` for is anything in `funnel_persons_query`.

Commit:
Update funnel success comment

Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>

neilkakkar reviewed (approved):
Query looks good to me, just waiting for tests.

Commit:
fix: filter events by team_id

Commit:
chore(correlation): remove CTEs from correlation query

There seems to be an issue with the CTEs and production clickhouse, see
https://github.com/ClickHouse/ClickHouse/issues/29748

Instead of risking it, I'm just removing them.

Commit:
chore: update entities -> events for funnel step exclusion

neilkakkar commented on ee/clickhouse/queries/funnels/funnel_correlation.py:
```diff
@@ -147,50 +147,112 @@ def get_partial_event_contingency_tables(self) -> List[EventContingencyTable]:
-            -- Right join, so we can get the total success/failure numbers as well
-            RIGHT JOIN funnel_people AS person
+            JOIN ({non_deleted_distinct_ids}) AS pdi
+                ON pdi.distinct_id = events.distinct_id AND pdi.team_id = event.team_id
```
The current test errors should go away after:
```suggestion
                ON pdi.distinct_id = events.distinct_id
```

Commit:
fix team_id = team_id issue
